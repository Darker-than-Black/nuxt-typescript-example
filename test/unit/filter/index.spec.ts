import vClickOutside from 'v-click-outside'
import BootstrapVue from 'bootstrap-vue'
import VueTestInstance from '~/test/factory/vue-instance'
import ExtendedOfFilters from '~/components/filter/index.vue'

describe('filter/management-table.vue', () => {
  let wrapper: any
  const value: any = {
    select: '',
    multiSelect: [],
    datalist: '',
    dateRange: []
  }

  beforeEach(() => {
    wrapper = VueTestInstance(ExtendedOfFilters, {
      modules: [BootstrapVue, vClickOutside],
      options: {
        propsData: { value }
      }
    })
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('check input function', () => {
    const { value, key }: { value: any, key: string } = { value: 'asd', key: 'key' }
    expect(wrapper.vm.input(value, key)).toEqual({ value, key })
  })
})
