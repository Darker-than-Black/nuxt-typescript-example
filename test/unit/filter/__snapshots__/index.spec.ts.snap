// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter/management-table.vue should match the snapshot 1`] = `
<div class="filters">
  <filter-multi-select-stub label="Multi select" options="[object Object],[object Object],[object Object],[object Object]" inputclasses="" value=""></filter-multi-select-stub>
  <filter-select-stub label="select" options="[object Object],[object Object],[object Object],[object Object]" inputclasses="" value=""></filter-select-stub>
  <filter-datalist-stub label="Datalist" options="Оберіть значення,Name 2,lorem ipsum,Name 4" inputclasses="" listlink="datalist" value=""></filter-datalist-stub>
  <datepickerrange-stub label="Date range" value=""></datepickerrange-stub>
</div>
`;
